src/pages/organizations/[organizationid].js

```diff
@@ -521,6 +521,11 @@ export async function getServerSideProps({ params }) {
   } catch (e) {
     log.warn('organizations page:', e);
     if (e.response?.status === 404) return { notFound: true };
-    throw e;
+    return {
+      redirect: {
+        destination: '/500',
+        permanent: false,
+      },
+    };
```

Comment:
Do you think we can use a function or so to reduce the boilerplate, like:

```
export async const getServersideProps = serverSiderPropsWrapper(() => {
...
})

function serverSiderPropsWrapper(callback){
   try{
       return callback();
    }catch(
....
}
```

Reply:
I'm not sure this is possible, but seems the code is a bit duplicated.

Reply:
Yes I was just think about this, we could turn this into a function! But what happens if for example in the tree page, we have the tree. But the organizations fails. Then we still get the 500 page, even with the previous way of handling the error

Reply:
I don't get it, what's `the organizations fails` ? 

I think we just extract the code and put into a function, so we can reuse it?

Reply:
Isn't the catch part (the code and logic) the same for every page?

Reply:
For example here:
https://github.com/Greenstand/treetracker-web-map-client/blob/beta/src/pages/trees/%5Btreeid%5D.js#L679-L689

What if the tree request is fine but the planter or organization request fails. Then the user will be redirected to the 500 page even if we have all the data from the tree itself.

Reply:
But I don't think this is relevant with the extracting-function-topic, right?
Back to your example, I think it's fine to return 500 if any requests failed in the page

Reply:
We'd better arrange a meeting for this if we can not agree with each other?

Reply:
If it is fine to return to the 500 if any request fails then its not a problem. I will refacter this to some wrapper for this!